3rdparty/fastcv/hal/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+set(HAL_LIB_NAME "fastcv_hal")
+
+set(FASTCV_HAL_FILES
+  fastcv_hal_core.cpp
+  fastcv_hal_core.hpp
+  fastcv_hal_imgproc.cpp
+  fastcv_hal_imgproc.hpp
+  fastcv_hal_utils.cpp
+  fastcv_hal_utils.hpp
+)
+
+add_library(fastcv_hal STATIC "${FASTCV_HAL_FILES}")
+target_include_directories(fastcv_hal PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${CMAKE_SOURCE_DIR}/modules/core/include
+  ${CMAKE_SOURCE_DIR}/modules/imgproc/include
+  fastcv_hal PRIVATE ${FASTCV_DIR}/inc)
+
+set(FASTCV_LIB "/libs/Android/lib64/libfastcvopt.so")
```

Comment:
Since there can be different builds of FastCV for different devices, should we handle it somehow?

Reply:
we will download different libraries to same folder for different platforms.